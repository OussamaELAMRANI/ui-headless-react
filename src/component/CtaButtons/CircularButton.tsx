import React, { FC } from "react";
import Button from "@/ui/Button";

export const CircularButton: FC<ButtonUiType> = ({
  size,
  variant,
  children,
  ...props
}) => {
  return (
    <Button
      withIcon
      variant={variant}
      size={size}
      {...props}
      className={"circular"}
      pill
    >
      {children}
    </Button>
  );
};
