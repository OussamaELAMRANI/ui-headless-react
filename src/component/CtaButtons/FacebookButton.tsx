import { FC } from "react";
import Button from "../../ui/Button/ss";
import { FacebookIcon } from "../../icon/FacebookIcon";

export const FacebookButton: FC = (prop) => {
  return (
    <Button
      withIcon
      variant={"primary"}
      className={"facebook text-2xl"}
      {...prop}
    >
      <>
        <FacebookIcon ss={""} />
        Facebook
      </>
    </Button>
  );
};
