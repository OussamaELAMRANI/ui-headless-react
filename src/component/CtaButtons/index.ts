export { FacebookButton } from "./FacebookButton";
export { CircularButton } from "./CircularButton";
