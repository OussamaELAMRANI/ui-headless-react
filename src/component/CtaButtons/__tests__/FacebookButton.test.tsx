import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";
import { FacebookButton } from "@/component/CtaButtons";

describe("FacebookButton", () => {
  afterEach(cleanup);

  it("should render correctly !", () => {
    const elem = create(<FacebookButton />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
