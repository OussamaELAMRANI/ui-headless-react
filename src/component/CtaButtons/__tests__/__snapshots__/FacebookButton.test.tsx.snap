// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx should render correctly ! 1`] = `
<button
  className="facebook"
  type="button"
>
  <span>
    <svg
      aria-hidden="true"
      data-icon="facebook-f"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.1 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.4 0 225.4 0c-73.22 0-121.1 44.38-121.1 124.7v70.62H22.89V288h81.39v224h100.2V288z"
        fill="currentColor"
      />
    </svg>
    Facebook
  </span>
</button>
`;

exports[`Buttons should render correctly ! 1`] = `
<button
  className="facebook"
  type="button"
>
  <span>
    <svg
      aria-hidden="true"
      data-icon="facebook-f"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.1 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.4 0 225.4 0c-73.22 0-121.1 44.38-121.1 124.7v70.62H22.89V288h81.39v224h100.2V288z"
        fill="currentColor"
      />
    </svg>
    Facebook
  </span>
</button>
`;

exports[`FacebookButton should render correctly ! 1`] = `
<button
  className="facebook"
  type="button"
>
  <span>
    <svg
      aria-hidden="true"
      data-icon="facebook-f"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.1 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.4 0 225.4 0c-73.22 0-121.1 44.38-121.1 124.7v70.62H22.89V288h81.39v224h100.2V288z"
        fill="currentColor"
      />
    </svg>
    Facebook
  </span>
</button>
`;
